src/test/spec/mod.rs

```diff
@@ -62,22 +62,26 @@ where
         }
 
         let test_file_full_path = base_path.join(&test_file_path);
-        let json: Value =
-            serde_json::from_reader(File::open(test_file_full_path.as_path()).unwrap()).unwrap();
+        run_single_test(test_file_full_path, &run_test_file).await;
+    }
+}
 
-        // Printing the name of the test file makes it easier to debug deserialization errors.
-        println!(
-            "Running tests from {}",
-            test_file_full_path.display().to_string()
-        );
+pub(crate) async fn run_single_test<T, F, G>(path: PathBuf, run_test_file: &F)
```

Comment:
The `valid_fail` tests need to be able to call this method directly to ensure that each test file in the directory fails; just calling `run_spec_test` will panic on the first test file and return.